components/media/PreviewMediaResolver.vue

```diff
@@ -0,0 +1,50 @@
+<template>
+  <MediaResolver
+    :src="properSrc"
+    :mimeType="properType"
+    :poster="poster"
+    preview />
+</template>
+
+<script lang="ts" setup>
+import { computed, defineAsyncComponent, onMounted, ref } from '#app'
+import { get, update } from 'idb-keyval'
+import axios from 'axios'
+
+import { NFTMetadata } from '@/components/rmrk/service/scheme'
+
+const MediaResolver = defineAsyncComponent(
+  () => import('@/components/media/MediaResolver.vue')
+)
+
+const props = defineProps({
+  src: { type: String, default: '' },
+  metadata: { type: String, default: '' },
+  mimeType: { type: String, default: '' },
+  poster: { type: String, default: '' },
+})
+
+const type = ref('')
+
+const properType = computed(() => props.mimeType || type.value || 'image/webp')
+const properSrc = computed(() => props.src || '/placeholder.webp')
+
+const fetchMimeType = async () => {
+  if (props.mimeType) {
+    return
+  }
+
+  const nftMetadata = await get<NFTMetadata>(props.metadata)
+  type.value = nftMetadata?.type || ''
+  if (!type.value) {
+    const { headers } = await axios.head(props.src)
```

Comment:
can't we [useFetch](https://v3.nuxtjs.org/guide/features/data-fetching/#usefetch) instead of axios?

Reply:
> Dolphins have returned to Italy.
> @roiLeo returned to the repo (welcome!)

Sure! let's do it

Reply:
sure, will check it 👍

Reply:
tried with `useLazyFetch` in this commit [`d23c5f9` (#3718)](https://github.com/kodadot/nft-gallery/pull/3718/commits/d23c5f9a297765baec0786c1a77eca15416f1b1a), because currently `useFetch` in nuxt-bridge is not available. but, the component will always request the URL even if the data exists in `idb-keyval`.

how about using `$fetch` instead? I already updated it 
- https://v3.nuxtjs.org/api/utils/$fetch/
- https://github.com/nuxt/framework/issues/1852#issuecomment-966161569